odoo/models.py

```diff
@@ -2449,7 +2449,13 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
                     row[group] = (value.id, value.sudo().display_name) if value else False
                     value = value.id
 
-                additional_domain = [(field_name, '=', value)]
+                if not value and field.type == 'many2many':
+                    other_values = [other_row[group][0] if isinstance(other_row[group], tuple)
+                                    else other_row[group].id if isinstance(value, BaseModel)
+                                    else other_row[group] for other_row in rows_dict if other_row[group]]
```

Comment:
> This makes the calculation quadratic. Should you just change `[(field_name, '=', False)]` to `[(field_name, 'not in', other_values)]` and leave `[(field_name, '=', <id>)]` as is ?

I'm not sure to get your point. Because this is exactly what I do if I understand well what you mean.

From what I understand, there should also be only one group whose `value` is `False`, therefore the list comprehension used to define `other_values` should only be executed for one iteration of the parent loop. So in practice it should not be quadratic, is it?

I guess an alternative could also be to explore all the groups' domain after the loop to replace the `(field_name, '=', False)` leaf but I'm not sure it could bring any performance gain and I think it would be more complex (in the sense less easily readable).

Reply:
Ho my bad, I missed the `not value and` in the if. Shame on me.